### PR TITLE
Localising and Lateralising KeyError when semiology not found (issue 7)

### PR DESCRIPTION
**Pull request by thenineteen**
excluded Localising and Lateralising columns from DataFrame dropna(), then fillna(0).

### CODE REVIEW

**Comment by thenineteen** [comment]
why does the build fail?!

**Comment by fepegar** [comment]
Can you use a more informative title for the PR? See my edit in the one you deleted earlier.

You can see the build log on Travis: https://travis-ci.com/github/thenineteen/Semiology-Visualisation-Tool/jobs/311628093

The best you can do from now on is testing locally before you commit, for example by running `pytest`. That way you don't need to wait for the Travis build to see that it breaks the library. You can keep pushing to this branch and tests will be triggered for every commit. Just please use informative titles for PRs and commits, for the sanity of our future selves.

File: mega_analysis/crosstab/mega_analysis/QUERY_SEMIOLOGY.py [hunk 1]
```diff
@@ -222,16 +222,33 @@ def QUERY_SEMIOLOGY(df, semiology_term=['love'],
...
+    inspect_result = \
+        inspect_result[[column for column in inspect_result.isna().all().index if column not in [keep]]].dropna(axis=1).append(inspect_result[[keep]], sort=False)
+    inspect_result = inspect_result.fillna(0)
 
     try:
         inspect_result.drop_duplicates(inplace=True)
     except ValueError:
         print('QUERY SEMIOLOGY ERROR: This semiology was not found within the reported literature nor in the semiology categories')
-        return
+        sys.exit()
```
Comment (fepegar) [inline mega_analysis/crosstab/mega_analysis/QUERY_SEMIOLOGY.py]: This is not a good idea. A function does not need to be able to finish the execution of the whole program.

File: mega_analysis/crosstab/mega_analysis/QUERY_SEMIOLOGY.py [hunk 2]
```diff
@@ -222,16 +222,33 @@ def QUERY_SEMIOLOGY(df, semiology_term=['love'],
...
+        logging.debug(f'\nLocalising Datapoints relevant to query {semiology_term}: {inspect_result["Localising"].sum()}')
+    except KeyError:
+        # user tried a semiology which doesn't have a key in semiology_dictionary
+        # run again and set semiology_dict to None
+    #     inspect_result = QUERY_SEMIOLOGY(
+    #         df,
+    #         semiology_term=semiology_term,
+    #         semiology_dict_path=None,
+    # )
+        logging.debug('Issue # 7 alive and well.')
```
Comment (fepegar) [inline mega_analysis/crosstab/mega_analysis/QUERY_SEMIOLOGY.py:248]: GitHub issues don't need to be referenced in the code. Also if code lines are commented out, they just shouldn't exist.

Comment (thenineteen) [reply mega_analysis/crosstab/mega_analysis/QUERY_SEMIOLOGY.py:247]: yes got rid of sys.exit(), still doesn't pass checks.

**Comment by thenineteen** [comment]
yayy

**Comment by thenineteen** [comment]
please check it resolves your issue #7

**Comment by fepegar** [comment]
> yayy

It's satisfying when builds work again, isn't it? :)

**Comment by fepegar** [comment]
Note that I squashed the commits into a single one (although I forgot to use a better name for it).

**Pull request merged**